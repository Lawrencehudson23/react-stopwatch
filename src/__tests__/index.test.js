
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import Stopwatch from '..';

Enzyme.configure({ adapter: new Adapter() });

const noop = () => undefined;

describe('ReactStopwatch', () => {
  it('It renders without breaking using render as property', () => {
    const wrapper = shallow(
      <Stopwatch
        seconds={0}
        minutes={0}
        hours={0}
        limit="00:00:10"
        withLoop
        // eslint-disable-next-line no-console
        onCallback={() => console.log('Finish')}
        render={noop}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('It renders without breaking using render as children elements', () => {
    const wrapper = shallow(
      <Stopwatch
        seconds={0}
        minutes={0}
        hours={0}
        limit="00:00:10"
        withLoop
        // eslint-disable-next-line no-console
        onCallback={() => console.log('Finish')}
        render={noop}
      >
        {() => <div />}
      </Stopwatch>,
    );

    expect(wrapper).toMatchSnapshot();
  });


  it('should start to works', (done) => {
    const wrapper = shallow(
      <Stopwatch
        seconds={0}
        minutes={0}
        hours={0}
        limit="00:00:10"
        withLoop
        // eslint-disable-next-line no-console
        onCallback={() => console.log('Finish')}
        render={noop}
      />,
    );

    expect(wrapper.state().stateHours).toBe(0);
    expect(wrapper.state().stateHours).toBe(0);
    expect(wrapper.state().stateMinutes).toBe(0);

    setTimeout(() => {
      expect(wrapper.state().stateSeconds).toBe(1);
      done();
    }, 1100);
  });

  it('should call to onCallback function', (done) => {
    const onFinish = () => {
      expect(true).toBe(true);
      done();
    };

    shallow(
      <Stopwatch
        seconds={0}
        minutes={0}
        hours={0}
        limit="00:00:03"
        withLoop={false}
        onCallback={onFinish}
        render={noop}
      />,
    );
  });

  it('should dont make a loop if withLoop is false ', (done) => {
    const wrapper = shallow(
      <Stopwatch
        seconds={0}
        minutes={0}
        hours={0}
        limit="00:00:02"
        withLoop={false}
        render={noop}
      />,
    );

    setTimeout(() => {
      expect(wrapper.state().stateSeconds).toBe(2);
      done();
    }, 2100);
  });

  it('should change to 1 minute', (done) => {
    const wrapper = shallow(<Stopwatch
      seconds={58}
      minutes={0}
      hours={0}
      limit="00:01:00"
      withLoop={false}
      render={noop}
    />);

    setTimeout(() => {
      expect(wrapper.state().stateMinutes).toBe(1);
      done();
    }, 2100);
  });

  it('should change to 1 hour', (done) => {
    const wrapper = shallow(
      <Stopwatch
        seconds={58}
        minutes={59}
        hours={0}
        limit="01:00:00"
        withLoop={false}
        render={noop}
      />,
    );

    setTimeout(() => {
      expect(wrapper.state().stateHours).toBe(1);
      done();
    }, 2100);
  });
});
