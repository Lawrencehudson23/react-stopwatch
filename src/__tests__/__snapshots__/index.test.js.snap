// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactStopwatch It renders without breaking using render as children elements 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReactStopWatch
    hours={0}
    limit="00:00:10"
    minutes={0}
    onCallback={[Function]}
    render={[Function]}
    seconds={0}
    withLoop={true}
  >
    [Function]
  </ReactStopWatch>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ReactStopwatch It renders without breaking using render as property 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReactStopWatch
    hours={0}
    limit="00:00:10"
    minutes={0}
    onCallback={[Function]}
    render={[Function]}
    seconds={0}
    withLoop={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
